src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,206 @@
+
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>();
+  const [filteredList, setFilteredList] = useState<Person[]>();
+  const [pageState, setPageState] = useState<PageState>('loading');
+  const [urlSlug, setUrlSlug] = useState<string>();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { slug } = useParams();
+
+  // Here it'll take the response from API and add father and mother to
+  // each person, if they exist...
+  const addMotherAndFather = (response: Person[]): Person[] => {
+    const peopleList = response.map((person) => {
+      const findMother = person.motherName
+        ? response.filter((p) => p.name === person.motherName)[0]
+        : null;
+
+      const findFather = person.fatherName
+        ? response.filter((p) => p.name === person.fatherName)[0]
+        : null;
+
+      return {
+        ...person,
+        ...(findFather && { father: findFather }),
+        ...(findMother && { mother: findMother }),
+      };
+    });
+
+    return peopleList;
+  };
+
+  const filter = (params: URLSearchParams) => {
+    const sex = params.get('sex');
+    const query = params.get('query');
+    const centuries = params.getAll('centuries');
+
+    setFilteredList(people);
+
+    if (sex) {
+      setFilteredList((current) => {
+        return current?.filter((person) => person.sex === sex);
+      });
+    }
+
+    if (query) {
+      setFilteredList((current) => current?.filter((person) => {
+        const searchQuery = query.toLocaleLowerCase();
+        const name = person.name.toLocaleLowerCase();
+        const motherName = String(person.motherName).toLocaleLowerCase();
+        const fatherName = String(person.fatherName).toLocaleLowerCase();
+
+        return (
+          name.includes(searchQuery)
+            || motherName.includes(searchQuery)
+            || fatherName.includes(searchQuery)
+        );
+      }));
+    }
+
+    if (centuries.length) {
+      setFilteredList((current) => current?.filter((person) => {
+        const personCentury = Math.ceil(person.born / 100);
+
+        return centuries.includes(String(personCentury));
+      }));
+    }
+  };
+
+  const sort = (params: URLSearchParams) => {
+    const sortBy = params.get('sort');
+    const orderBy = params.get('order');
+    const sortByText = sortBy === 'name' || sortBy === 'sex';
+
+    if (sortBy && orderBy === 'desc') {
+      setFilteredList((current) => current?.sort((a: Person, b: Person) => {
+        if (sortByText) {
+          return a[sortBy].localeCompare(b[sortBy]);
+        }
+
+        if (sortBy === 'born' || sortBy === 'died') {
+          return a[sortBy] - b[sortBy];
+        }
+
+        return 1;
+      }));
+    }
+
+    if (sortBy && !orderBy) {
+      setFilteredList((current) => current?.sort((a: Person, b: Person) => {
+        if (sortByText) {
+          return b[sortBy].localeCompare(a[sortBy]);
+        }
+
+        if (sortBy === 'born' || sortBy === 'died') {
```

Comment:
and here you could reuse the `const sortByInteger`
```suggestion
        if (sortByInteger) {
```